kotlin/internal/jvm/compile.bzl

```diff
@@ -101,16 +102,26 @@ def _jvm_deps(toolchains, associated_targets, deps, runtime_deps = []):
             ",\n ".join(["    %s" % x for x in list(diff)]),
         )
     dep_infos = [_java_info(d) for d in associated_targets + deps] + [toolchains.kt.jvm_stdlibs]
+
+    # Reduced classpath, exclude transitive deps from compilation
+    if (ctx.attr._experimental_prune_transitive_deps[BuildSettingInfo].value):
```

Comment:
Would it be better to expose this at the `toolchains.kt` level so that there's a unified place to look up these high level build settings?

Reply:
You're talking about `kotlin/internal/toolchains.bzl` right? defining this in toolchains will prevent you from passing this flag on CLI, no?
Right now you're able to do something like this:
```
bazel build --@io_bazel_rules_kotlin//kotlin/internal/jvm:experimental_prune_transitive_deps=True //foo
```

Reply:
Ah yes you are actually right.

Reply:
That isn't strictly true right? You can always have a config setting that enables the feature in the tool chain.

We do this for conditional multiplex workers 

```
config_setting( 
     name = "kotlinc_multiplex_workers", 
     define_values = { 
         "kotlinc_multiplex_workers": "enabled", 
     }, 
 ) 
  
 define_kt_toolchain( 
     name = "toolchain", 
     api_version = "1.7", 
     experimental_multiplex_workers = select({ 
         ":kotlinc_multiplex_workers": True, 
         "//conditions:default": False, 
     }),
```

Then enable it via `--define=kotlinc_multiplex_workers=enabled` one could also do this as a proper flag. The benefits of this is that we don't need to ship skylib in the release archive and downstream users get full control over how it's toggled and set.

Reply:
You're right, but that requires a lot of setup on the user side that can potentially be error prone. I personally think that having it via skylib is more user friendly and aligns with all other first class citizen flags we support. Though I don't have super strong feeling either way.

Reply:
you can attach the build flag on the toolchain rule, there's no need for this build flag to be an attr for pretty much every kotlin rule...

Reply:
As Cris recommended, I've moved the flag under `kotlin/settings` and attached it to the toolchain as it's cleaner that way. 
I'll do the same for the jdeps, `kotlin_deps` flag for consistency in a follow up pr.